src/python/WMCore/WMSpec/Steps/Executors/DQMUpload.py

```diff
@@ -293,3 +309,229 @@ def upload(self, url, args, filename):
+
+        return eosFileInfo['LFN']
+
+    def getAnalysisFileLFN(self, analysisFile):
+        """
+        Construct an LFN for an analysisFile
+        """
+        # If lfn start with '/', make it relative to it
+        lfnFile = os.path.relpath(analysisFile.lfn, start='/')
+        lfn = os.path.join(self.registerLFNBase, lfnFile)
+    
+        return lfn
+
+    def _register(self, registerURL, args):
+        """
+        POST request to register URL
+        """
+
+        ident = "WMAgent python/%d.%d.%d" % sys.version_info[:3]
+        uploadProxy = self.step.upload.proxy or os.environ.get('X509_USER_PROXY', None)
+        logging.info("Using proxy file: %s", uploadProxy)
+        logging.info("Using CA certificate path: %s", os.environ.get('X509_CERT_DIR'))
+
+        msg = "HTTP Register POST arguments: %s\n" % args
+        logging.info(msg)
+
+        handler = HTTPSAuthHandler(key=uploadProxy, cert=uploadProxy)
+        opener = OpenerDirector()
+        opener.add_handler(handler)
+
+        # setup the request object
+        url = decodeBytesToUnicode(registerURL)
+        datareq = Request(url)
+        datareq.add_header('Accept-encoding', 'gzip')
+        datareq.add_header('User-agent', ident)
+        datareq.data = encodeUnicodeToBytes(json.dumps(args))
+
+        if 'https://' in url:
+            result = opener.open(datareq)
+        else:
+            opener.add_handler(ProxyHandler({}))
+            result = opener.open(datareq)
+
+        return result
+    
+    def _registerAnalysisFile(self, stepName, analysisFile):
+        """
+        Register a file to new DQM GUi
+        - https://github.com/cms-DQM/dqmgui
+        HTTP request body:
+            [{"dataset": "/a/b/c", "run": "123456", "lumi": "0", "file": "/eos/cms/store/group/comm_dqm/DQMGUI_data/location/file.root", "fileformat": 1}]
+        Set lumis to 0, as this is reproducing current per Run based root files
+        - https://github.com/dmwm/WMCore/issues/10287#issuecomment-1052558061
+        DQM root files produced by Harvesting prosessing are type 1 (plain ROOT)
+        - https://github.com/cms-DQM/dqmgui#file-formats
+        """
+        # Get task description
+        job = Bootstrap.loadJobDefinition()
+        jobBag = job.getBaggage()
+        datasetName = job.get('inputDataset', None)
+
+        # Get runNumber, depending on the harvesting job mode (byRun vs multiRun)
+        multiRun = getattr(jobBag, "multiRun", False)
+        forceRunNumber = getattr(jobBag, "forceRunNumber", 999999)
+        if multiRun:
+            runNumber = forceRunNumber
+        else:
+            try:
+                runNumber = list(job['mask']['runAndLumis'].keys())[0]
+            except Exception as ex:
+                msg = "Error while retrieving run number from job definition:\n %s" % str(ex)
+                raise WMExecutionFailure(70320, "DQMUploadFailure", msg)    
+
+        args = {}
+        args['file'] = self.registerEOSPrefix + \
```

Comment:
It might be safer to use `os.path.join` to avoid issues with double or lack of slashes. The line looks short enough as well, maybe put it all in the same line.

Reply:
Same as one comment above, I would need to remove the starting `/` from the second argument in order to use `os.path.join`, otherwise we get:

```
>>> os.path.join("/eos/cms","/store/user/something")
'/store/user/something'
```